tests/tracer/test_span.py

```diff
@@ -676,3 +676,15 @@ def test_manual_context_usage():
     span1.context.sampling_priority = 1
     assert span2.context.sampling_priority == 1
     assert span1.context.sampling_priority == 1
+
+
+def test_set_exc_info_with_unicode():
+    exception_message = u"DataDog/水"
+    span = Span(None, "span1")
+    try:
+        raise Exception(exception_message)
+    except Exception:
+        type_, value_, traceback_ = sys.exc_info()
+        span.set_exc_info(type_, value_, traceback_)
```

Comment:
I wonder if it would be more accurate to do as:

```
with pytest.raises(Exception):
    with Span(None, "span1") as span:
        raise Exception(u"...")
```

Not a strong opinion, just maybe different approach.

Reply:
This format works but it's a little harder to follow: 

```
def test_set_exc_info_with_unicode():
    def get_exception_span(message):
        span = None
        try:
            with Span(None, "span1") as span:
                raise Exception(message)
        except Exception:
            pass
        return span

    exception_span = get_exception_span(u"DataDog/水")
    assert u"DataDog/水" == exception_span.get_tag(ERROR_MSG)

    if six.PY3:
        exception_span = get_exception_span("DataDog/水")
        assert "DataDog/水" == exception_span.get_tag(ERROR_MSG)
```